mne/decoding/time_gen.py

```diff
@@ -625,39 +707,34 @@ def _sliding_window(times, window_params, sfreq):
         List of training slices, indicating for each classifier the time
         sample (in indices of times) to be fitted on.
     """
+    import copy
 
-    window_params = _DecodingTime(window_params)
+    window = _DecodingTime(copy.deepcopy(window))
 
     # Default values
-    if ('slices' in window_params and
-            all(k in window_params for k in
-                ('start', 'stop', 'step', 'length'))):
-        time_pick = window_params['slices']
+    has_slice = 'slices' in window
+    if has_slice:
+        time_pick = window['slices']
```

Comment:
I don't understand this code in general.
What it's currently doing is:
```
if 'slices' in window:
    time_pick = window['slices']
else:
    window['start'] = window.get('start', times[0])
    window['stop'] = window.get('stop', times[-1])
....
```

right?

And if 'slices' is indeed in `window`, it sets `time_pick` and then immediately jumps to line 756, never touching `time_pick` ever again?

Shouldn't it be

```
if not "slices" in window:
        window['start'] = window.get('start', times[0])
...
```

Or am I missing something?

Also, the doc currently says it returns `time_pick`, but it returns `window`.